drivers/vmwarevsphere/create.go

```diff
@@ -0,0 +1,413 @@
+	}
+
+	var add []types.BaseVirtualDevice
+
+	controller, err := devices.FindDiskController("scsi")
+	if err != nil {
+		return err
+	}
+
+	disk := devices.CreateDisk(controller, dss.Reference(),
+		dss.Path(fmt.Sprintf("%s/%s.vmdk", d.MachineName, d.MachineName)))
+
+	// Convert MB to KB
+	disk.CapacityInKB = int64(d.DiskSize) * 1024
+
+	add = append(add, disk)
+	ide, err := devices.FindIDEController("")
+	if err != nil {
+		return err
+	}
+
+	cdrom, err := devices.CreateCdrom(ide)
+	if err != nil {
+		return err
+	}
+
+	add = append(add, devices.InsertIso(cdrom, dss.Path(fmt.Sprintf("%s/%s", d.MachineName, isoFilename))))
+
+	if vm.AddDevice(d.getCtx(), add...); err != nil {
+		return err
+	}
+
+	if err := d.addNetworks(vm, d.networks); err != nil {
+		return err
+	}
+
+	if err := d.addConfigParams(vm); err != nil {
+		return err
+	}
+
+	if err := d.cloudInit(vm); err != nil {
+		return err
+	}
+
+	if err := d.Start(); err != nil {
+		return err
+	}
+
+	if err := d.provisionVm(vm); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (d *Driver) createFromVmName() error {
+	c, err := d.getSoapClient()
+	if err != nil {
+		return err
+	}
+
+	dss, err := d.finder.DatastoreOrDefault(d.getCtx(), d.Datastore)
+	if err != nil {
+		return err
+	}
+
+	var info *types.TaskInfo
+	ref := d.resourcepool.Reference()
+	spec := types.VirtualMachineCloneSpec{
+		Location: types.VirtualMachineRelocateSpec{
+			Pool: &ref,
+		},
+		Config: &types.VirtualMachineConfigSpec{
+			GuestId:  "otherLinux64Guest",
+			Files:    &types.VirtualMachineFileInfo{VmPathName: fmt.Sprintf("[%s]", dss.Name())},
+			NumCPUs:  int32(d.CPU),
+			MemoryMB: int64(d.Memory),
+		},
+	}
+
+	vm2Clone, err := d.fetchVM(d.CloneFrom)
+	if err != nil {
+		return err
+	}
+
+	folders, err := d.datacenter.Folders(d.getCtx())
+	if err != nil {
+		return err
+	}
+
+	task, err := vm2Clone.Clone(d.getCtx(), folders.VmFolder, d.MachineName, spec)
+	if err != nil {
+		return err
+	}
+
+	info, err = task.WaitForResult(d.getCtx(), nil)
+	if err != nil {
+		return err
+	}
+
+	// Retrieve the new VM
+	vm := object.NewVirtualMachine(c.Client, info.Result.(types.ManagedObjectReference))
+	if err := d.addNetworks(vm, d.networks); err != nil {
+		return err
+	}
+
+	if err := d.addConfigParams(vm); err != nil {
+		return err
+	}
+
+	if err := d.cloudInit(vm); err != nil {
+		return err
+	}
+
+	if err := d.Start(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (d *Driver) createFromLibraryName() error {
+	c, err := d.getSoapClient()
+	if err != nil {
+		return err
+	}
+
+	ds, err := d.finder.DatastoreOrDefault(d.getCtx(), d.Datastore)
+	if err != nil {
+		return err
+	}
+
+	folders, err := d.datacenter.Folders(d.getCtx())
+	if err != nil {
+		return err
+	}
+
+	manager, err := d.restLogin(d.getCtx(), c.Client)
+	if err != nil {
+		return err
+	}
+
+	results, err := vapifinder.NewFinder(manager).Find(d.getCtx(), d.CloneFrom)
+	if err != nil {
+		return err
+	}
+
+	if len(results) != 1 {
```

Comment:
should this be < 1?

Reply:
i need it to be one, it's a finder query with a name being passed and i dont want to accidently get two results some accidently. so i will rewrite the error as < 0 nothing found err, 1 > more than one err